src/connection/coniql.ts

```diff
@@ -497,7 +497,9 @@ export class ConiqlPlugin implements Connection {
     // Note that connectionMiddleware handles multiple subscriptions
     // for the same PV at present, so if this method is called then
     // there is no further need for this PV.
-    this.subscriptions[pvName].unsubscribe();
-    delete this.subscriptions[pvName];
+    if (this.subscriptions[pvName]) {
```

Comment:
Javascript doesn't generate an exception when reading a property that doesn't exist - it just returns `undefined`. Indeed the error I was originally seeing was: 
`Error: TypeError: this.subscriptions[pvName] is undefined`

So the above check will return `undefined` if that pv does not exist in the dictionary. `undefined` is falsy in Javascript and so this it will evaluate as false if they key does not exist and true otherwise. 
I could make this more explicit if you think it would help, e.g. check `this.subscriptions[pvname] !== undefined`?

Reply:
No it's fine as is, thanks, just showing the limits of my knowledge